scripts/upgrade

```diff
@@ -34,21 +34,75 @@ if ! groups $app | grep -q 'ssl-cert'; then
         adduser $app ssl-cert
 fi
 
+#=================================================
+# GET CONFIG PANEL SETTINGS
+#=================================================
+
+server_name=$(ynh_app_setting_get --app=$app --key=server_name)
+domain=$(ynh_app_setting_get --app=$app --key=domain)
+
+registration_disabled=$(ynh_app_setting_get --app=$app --key=registration_disabled)
+disable_federation=$(ynh_app_setting_get --app=$app --key=disable_federation)
+guests_disabled=$(ynh_app_setting_get --app=$app --key=guests_disabled)
+registration_shared_secret=$(ynh_app_setting_get --app=$app --key=registration_shared_secret)
+enable_registration_captcha=$(ynh_app_setting_get --app=$app --key=enable_registration_captcha)
+
+#=================================================
+# STANDARD UPGRADE STEPS
+#=================================================
+# MIGRATION : Manage old settings
+#=================================================
+
+
 # Define $server_name if not already defined
-if [ -z $server_name ]; then
+if [ -z ${server_name:-} ]; then
    server_name=$domain
    ynh_app_setting_set --app=$app --key=server_name --value=$domain
 fi
 
+# Define $disable_federation if not already defined
+if [ -z ${enable_registration_captcha:-} ]; then
+   enable_registration_captcha="false"
+   ynh_app_setting_set --app=$app --key=enable_registration_captcha --value=$enable_registration_captcha
+fi
+
+# Define $guests_disabled if not already defined
+if [ -z ${guests_disabled:-} ]; then
+   guests_disabled="true"
+   ynh_app_setting_set --app=$app --key=guests_disabled --value=$guests_disabled
+fi
+
+# Define $registration_shared_secret if not already defined
+if [ -z ${registration_shared_secret:-} ]; then
+   registration_shared_secret=""
+   ynh_app_setting_set --app=$app --key=registration_shared_secret --value=$registration_shared_secret
+fi
+
 # Load up registration variables
-if [[ $registration -eq 1 ]]
-then
-        registration_disabled="false"
-        really_enable_open_registration="--really-enable-open-registration"
+registration=$(ynh_app_setting_get --app=$app --key=registration)
+if [ -z ${registration:-} ]; then
+	if [[ $registration_disabled = *"alse" ]]
+	then
+		really_enable_open_registration="--really-enable-open-registration"
+	else
+		really_enable_open_registration=""
+	fi
 else
-        registration_disabled="true"
-        really_enable_open_registration=""
-        ynh_app_setting_set --app=$app --key=registration --value=0
+	ynh_app_setting_delete --app=$app --key=registration
+	if [[ $registration == *"rue" || $registration -eq 1 ]]
```

Comment:
same here `*"rue"`

Reply:
No. I discovered that somehow a "True" was stored in the settings file of YunoHost, instead of "true". So I'm covering all cases here.